gax-java/gax-httpjson/src/main/java/com/google/api/gax/httpjson/ProtoMessageRequestFormatter.java

```diff
@@ -95,10 +95,28 @@ public String getRequestBody(RequestT apiMessage) {
     return requestBodyExtractor.extract(apiMessage);
   }
 
-  /* {@inheritDoc} */
+  /**
+   * Returns the relative URL path created from the path parameters from the given message. Attempts
+   * to match the with the default PathTemplate. If there is not match, it attempts to match with
+   * the templates in the additionalPathTemplates.
+   *
+   * @param apiMessage Request object to extract fields from
+   * @return Path of a matching URL
+   */
   @Override
   public String getPath(RequestT apiMessage) {
-    return pathTemplate.instantiate(pathVarsExtractor.extract(apiMessage));
+    String path = pathTemplate.instantiate(pathVarsExtractor.extract(apiMessage));
+    if (pathTemplate.matches(path)) {
+      return path;
+    }
+    for (PathTemplate additionalPathTemplate : additionalPathTemplates) {
+      String additionalPath =
+          additionalPathTemplate.instantiate(pathVarsExtractor.extract(apiMessage));
```

Comment:
Is `pathVarsExtractor.extract(apiMessage)` expected to return different values each time?

If not, please save the result to a local variable in the first line of this method rather than repeating this call.

Reply:
Ah, will update! I was prototyping with part 2 which has `pathVarsExtractor.extract(apiMessage)` return different values each time.